db/src/urls.rs

```diff
@@ -69,19 +73,29 @@ async fn download_raw_html(url: &str) -> color_eyre::Result<String> {
     Ok(html)
 }
 
+fn clean_raw_html(raw_html: &str, url: &Url) -> color_eyre::Result<String> {
+    let mut raw_html_cursor = std::io::Cursor::new(raw_html);
+    let article = extractor::extract(&mut raw_html_cursor, url)
+        .map_err(|e| color_eyre::eyre::eyre!(e.to_string()))?;
+    Ok(article.content)
+}
+
 async fn store_raw_html_in_page_snapshot(
     pool: &PgPool,
     page: Page,
 ) -> color_eyre::Result<PageSnapShot> {
     let raw_html = download_raw_html(&page.url).await?;
     let current_time = chrono::Utc::now();
+    let url = Url::parse(&page.url)?;
```

Comment:
used the `url` crate to parse the URL String into a URL a object. This is necessary because the `clean_raw_html` requires a url object as one of the function arguments.

By parsing the URL string into a URL string we ensure the URL that the user inserted is valid.